generator-maven-plugin/README.adoc

```diff
@@ -0,0 +1,264 @@
+JUDO PSM Generator maven plugin
+===============================
+
+This plugin manages and executes generators for JUDO PSM models.
+
+It generates text based codes / reports from PSM model with the given templates.
+
+== Requirements
+
+- Maven 3.8.3 and Java 11
+
+== Installation
+
+Include the plugin as a dependency in your Maven project. Change `LATEST_VERSION` to the latest tagged version.
+
+```
+<plugin>
+    <groupId>hu.blackbelt.judo.meta</groupId>
+	<artifactId>judo-psm-generator-maven-plugin</artifactId>
+    <version>LATEST_VERSION</version>
+    ...
+</plugin>
+...
+
+```
+
+== Usage
+
+Executing a template for an PSM model
+
+```
+
+<plugin>
+    <groupId>hu.blackbelt.judo.meta</groupId>
+    <artifactId>judo-psm-generator-maven-plugin</artifactId>
+    <version>${judo-psm-archetype-version}</version>
+    <executions>
+        <execution>
+            <id>execute-psm-test-model-from-artifact</id>
+            <phase>test</phase>
+            <goals>
+                <goal>generate</goal>
+            </goals>
+            <configuration>
+                <psm> <!-- 1 -->
+                    mvn:hu.blackbelt.judo.tatami:judo-tatami-northwind-psm:${judo-tatami-version}!model/northwind-psm.model
+                </psm>
+                <uris> <!-- 2 -->
+                    <uri>mvn:hu.blackbelt.judo.meta:judo-psm-fullstack-project-archetype:${judo-psm-fullstack-project-archetype-version}</uri>
+                    <uri>${basedir}/src/main/resources</uri>
+                </uris>
+                <helpers> <!-- 3 -->
+                    <helper>hu.blackbelt.judo.psm.fullstack.project.archetype.PsmProjectHelper</helper>
+                </helpers>
+                <type>fullstack-project</type> <!-- 4 -->
+                <destination>${basedir}/target/test-classes/psm/artifact</destination> <!-- 5 -->
+                <templateParameters> <!-- 6 -->
+                    <judoPlatformVersion>${judo-platform-version}</judoPlatformVersion>
+                </templateParameters>
+                <contextAccessor>hu.blackbelt.judo.psm.fullstack.project.archetype.ActorTypeValueResolver</contextAccessor> <!-- 7 -->
+                <scanDependencies>true</scanDependencies> <!-- 8 -->
+                <actors></actors> <!-- 9 -->
+            </configuration>
+        </execution>
+    </executions>
+
+    <dependencies>
+        <dependency>
+            <groupId>hu.blackbelt.judo.meta</groupId>
```

Comment:
We should add comments / notes stating that these dependencies are just example models for testing.